configstack/running_module.go

```diff
@@ -102,23 +109,34 @@ func RunModulesWithHandler(modules []*TerraformModule, handler ModuleHandler, or
 		return err
 	}
 
-	var waitGroup sync.WaitGroup
-	initSlowDown() // starts mechanism that control the maximum number of threads launched simultaneously
+	for _, module := range runningModules {
+		// Starts mechanism that control the maximum number of active workers
+		if module.Module.TerragruntOptions.NbWorkers <= 0 {
+			module.Module.TerragruntOptions.NbWorkers = len(runningModules)
+		}
+		initWorkers(module.Module.TerragruntOptions.NbWorkers)
+		break
+	}
 
+	var waitGroup sync.WaitGroup
 	for _, module := range runningModules {
 		waitGroup.Add(1)
 		module.Handler = handler
 		go func(module *runningModule) {
-			defer waitGroup.Done()
+			var completed bool
+			defer func() {
+				waitGroup.Done()
+				completed = true
+			}()
 			logCatcher := util.LogCatcher{
 				Writer: &module.OutStream,
 				Logger: module.Module.TerragruntOptions.Logger,
 			}
 			module.Module.TerragruntOptions.Writer = logCatcher
 			module.Module.TerragruntOptions.ErrWriter = logCatcher
-			var completed bool
 			go module.OutputPeriodicLogs(&completed) // Flush the output buffers periodically to confirm that the process is still alive
 			module.runModuleWhenReady()
+			completed = true
```

Comment:
Do we still need this line if it is now in the defer func?

Reply:
No, you are right!